File: 2086_minimum_number_of_food_buckets_to_feed_the_hamsters/solution.go
// LeetCode 2086 - Minimum Number of Food Buckets to Feed the Hamsters
// https://leetcode.com/problems/minimum-number-of-food-buckets-to-feed-the-hamsters/

func minimumBuckets(hamsters string) int {
	b := []byte(hamsters)
	ans := 0
	for i := 0; i < len(b); i++ {
		if b[i] != 'H' {
			continue
		}
		if i > 0 && b[i-1] == 'B' {
			continue
		}
		if i+1 < len(b) && b[i+1] == '.' {
			b[i+1] = 'B'
			ans++
		} else if i > 0 && b[i-1] == '.' {
			b[i-1] = 'B'
			ans++
		} else {
			return -1
		}
	}
	return ans
}

File: 2086_minimum_number_of_food_buckets_to_feed_the_hamsters/solution.js
// LeetCode 2086 - Minimum Number of Food Buckets to Feed the Hamsters
// https://leetcode.com/problems/minimum-number-of-food-buckets-to-feed-the-hamsters/

/**
 * @param {string} hamsters
 * @return {number}
 */
var minimumBuckets = function(hamsters) {
    const b = hamsters.split('');
    let ans = 0;
    for (let i = 0; i < b.length; i++) {
        if (b[i] !== 'H') continue;
        if (i > 0 && b[i - 1] === 'B') continue;
        if (i + 1 < b.length && b[i + 1] === '.') { b[i + 1] = 'B'; ans++; }
        else if (i > 0 && b[i - 1] === '.') { b[i - 1] = 'B'; ans++; }
        else return -1;
    }
    return ans;
};
